### PR TITLE
Iss86 resource manager integration

### PR DESCRIPTION
**Pull request by sandrain**
### Description
This includes a basic implementation of the unifier command line utility, addressing issue #86 .

### Motivation and Context
See issue #86.

### Current status
It successfully read configurations from sysconf file (/etc/unifycr/unifycr.conf), environmental variables, and command line arguments. Also, it generates the dynamic configuration file under specified path (runstatedir/unifycr/unifycr-runtime.conf, which can be changed by sysconf file). It is not launching the daemon, yet. In specific, the following is the remaining work list:

1. Launching the server daemon (do we need to execute mpirun and daemonize?)
2. Node list parsing using scontrol (SLURM), or manually.
3. Plugins for individual resource managers.
4. Changes to the unifycrd (server daemon) to read configurations from the dynamically created runtime configuration file.

The test program is going to be included after the item 4. above (changing the unifycrd).

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the UnifyCR code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted.

### CODE REVIEW

**Comment by nedbass** [comment]
> I checked the build log of travis, and it is not based on my latest commit. Is this a known issue?

@sandrain it ran on commit 21163c5 which appears to be the latest. Is that not what you expect? Travis fetches a merge of your PR branch and tests that (`git fetch origin +refs/pull/102/merge: ; git checkout FETCH_HEAD`). So the commit hash you see in the travis logs will be different from your latest commit.

**Comment by nedbass** [comment]
This line has a binary character in it that tricks github and the `file` command into thinking it's a binary file. I think it's supposed to be a dash.

https://github.com/sandrain/UnifyCR/blob/21163c59bbbea3ce2fbeacdf0686305c1d3eccd2/util/unifycr/src/toml.c#L147

**Comment by nedbass** [comment]
@sandrain since toml.[ch] aren't really part of UnifyCR, you can exempt them from the style checker by adding this line to your commit message:

```
TEST_CHECKPATCH_SKIP_FILES=util/unifycr/src/toml.c,util/unifycr/src/toml.h
```

**Comment by nedbass** [comment]
@sandrain now I seem what you mean about the latest commit not getting tested. It's because your patch stack contains merge commits and the style checking script doesn't handle that. It would work better if you squashed your work into a single commit and rebased it on the current dev branch.

**Comment by sandrain** [comment]
Okay. I will fix those things and will make a request again.

File: util/unifycr/src/unifycr-env.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * For details, see https://github.com/llnl/burstfs
+ * Please read https://github.com/llnl/burstfs/LICENSE for full license text.
+ */
+
+/*
+ *
+ * Copyright (c) 2014, Los Alamos National Laboratory
+ *	All rights reserved.
+ *
+ */
```
[the remark this answers is not in the document]
Comment (sandrain) [reply util/unifycr/src/unifycr-env.c]: I have just copied from other source files. Let's fix this issue globally later.

**Comment by sandrain** [comment]
I am closing this and creating a new PR with applying all fixes suggested by @nedbass .

**Pull request closed without merging**